sunscreen_tfhe/src/ops/encryption/glev_encryption.rs

```diff
@@ -0,0 +1,271 @@
+use crate::{
+    dst::FromMutSlice,
+    entities::{GlevCiphertextRef, GlweCiphertextRef, GlweSecretKeyRef, Polynomial, PolynomialRef},
+    polynomial::polynomial_scalar_mul,
+    scratch::allocate_scratch_ref,
+    GlweDef, RadixDecomposition, Torus, TorusOps,
+};
+
+use super::{
+    decrypt_glwe_ciphertext, encrypt_glwe_ciphertext_secret,
+    trivially_encrypt_glwe_with_sk_argument,
+};
+
+/// Perform a GLev encryption. This is generic in case a trivial GLev encryption
+/// is wanted (for example, for testing purposes).
+pub(crate) fn encrypt_glev_ciphertext_generic<S>(
```

Comment:
I think it's okay to export these publicly. Maybe some cryptographers wanna do some weird stuff with GLEVs?